ext4: Add i_version_hi field for ext4_inode

From: Kalpak Shah <kalpak@clusterfs.com>

This patch adds a 32-bit i_version_hi field to ext4_inode,
which can be used for 64-bit inode versions. This field will
store the higher 32 bits of the version, while Jean Noel's patch
has added support to store the lower 32-bits in osd1.linux1.l_i_version.

Signed-off-by: Mingming Cao <cmm@us.ibm.com>
Signed-off-by: Andreas Dilger <adilger@clusterfs.com>
Signed-off-by: Kalpak Shah <kalpak@clusterfs.com>
---

 include/linux/ext4_fs.h |    1 +
 1 file changed, 1 insertion(+)


Index: linux-2.6.23-rc5/include/linux/ext4_fs.h
===================================================================
--- linux-2.6.23-rc5.orig/include/linux/ext4_fs.h	2007-09-12 16:13:18.000000000 -0700
+++ linux-2.6.23-rc5/include/linux/ext4_fs.h	2007-09-12 16:16:37.000000000 -0700
@@ -349,6 +349,7 @@ struct ext4_inode {
 	__le32  i_atime_extra;  /* extra Access time      (nsec << 2 | epoch) */
 	__le32  i_crtime;       /* File Creation time */
 	__le32  i_crtime_extra; /* extra FileCreationtime (nsec << 2 | epoch) */
+	__le32	i_version_hi;	/* high 32 bits for 64-bit version */
 };
 
 #define i_size_high	i_dir_acl
